apis/vshn/v1/dbaas_vshn_redis.go

```diff
@@ -0,0 +1,101 @@
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of a VSHNRedis.
+	Spec VSHNRedisSpec `json:"spec"`
+
+	// Status reflects the observed state of a VSHNRedis.
+	Status VSHNRedisStatus `json:"status,omitempty"`
+}
+
+// VSHNRedisSpec defines the desired state of a VSHNRedis.
+type VSHNRedisSpec struct {
+	// Parameters are the configurable fields of a VSHNRedis.
+	Parameters VSHNRedisParameters `json:"parameters,omitempty"`
+}
+
+// VSHNRedisParameters are the configurable fields of a VSHNRedis.
+type VSHNRedisParameters struct {
+	// Service contains Redis DBaaS specific properties
+	Service VSHNRedisServiceSpec `json:"service,omitempty"`
+
+	// Size contains settings to control the sizing of a service.
+	Size VSHNRedisSizeSpec `json:"size,omitempty"`
+
+	// Backup contains settings to control the backups of an instance.
+	Backup VSHNRedisBackupSpec `json:"backup,omitempty"`
+}
+
+// VSHNRedisServiceSpec contains Redis DBaaS specific properties
+type VSHNRedisServiceSpec struct {
+	// +kubebuilder:validation:Enum="6.2";"7.0"
+	// +kubebuilder:default="7.0"
+
+	// Version contains supported version of Redis.
+	// Multiple versions are supported. The latest version "7.0" is the default version.
+	Version string `json:"version,omitempty"`
+
+	// RedisSettings contains additional Redis settings.
+	RedisSettings string `json:"redisSettings,omitempty"`
+}
+
+// VSHNRedisSizeSpec contains settings to control the sizing of a service.
+type VSHNRedisSizeSpec struct {
+	// +kubebuilder:default="500m"
+
+	// CPURequests defines the requests amount of Kubernetes CPUs for an instance.
+	CPURequests string `json:"cpuRequests,omitempty"`
+
+	// +kubebuilder:default="500m"
+
+	// CPULimits defines the limits amount of Kubernetes CPUs for an instance.
+	CPULimits string `json:"cpuLimits,omitempty"`
+
+	// +kubebuilder:default="2048Mi"
+
+	// MemoryRequests defines the requests amount of memory in units of bytes for an instance.
+	MemoryRequests string `json:"memoryRequests,omitempty"`
+
+	// +kubebuilder:default="2048Mi"
+
+	// MemoryLimits defines the limits amount of memory in units of bytes for an instance.
+	MemoryLimits string `json:"memoryLimits,omitempty"`
+
+	// +kubebuilder:default="5Gi"
+
+	// Disk defines the amount of disk space for an instance.
+	Disk string `json:"disk,omitempty"`
+}
+
+// VSHNRedisBackupSpec contains settings to control the backups of an instance.
+type VSHNRedisBackupSpec struct {
+	// +kubebuilder:validation:Pattern="^([0-1]?[0-9]|2[0-3]):([0-5][0-9]):([0-5][0-9])$"
+	// +kubebuilder:default="21:30:00"
+
+	// TimeOfDay for doing daily backups, in UTC.
+	// Format: "hh:mm:ss".
+	TimeOfDay string `json:"timeOfDay,omitempty"`
+}
+
+// VSHNRedisStatus reflects the observed state of a VSHNRedis.
+type VSHNRedisStatus struct {
+	// RedisConditions contains the status conditions of the backing object.
+	RedisConditions []v1.Condition `json:"redisConditions,omitempty"`
```

Comment:
I don't see a patch in the composition that would actually fill in these conditions.

Reply:
True, since we haven't implemented anything yet. However, we will probably need it soon to get some events forwarded to the claim. Should I remove it for now or leave it?

Reply:
Leaving it in without populating it could lead to confusions... I'd remove it until it's properly added.

Reply:
Ok. removed it as well